### ISSUE TITLE
Lifetime problem

### ISSUE THREAD

**Issue by crone123** [sunng87/handlebars-rust#283]
Hello,
I want to do a condition with handlebars as an helper, like the second post of this thread, but in rust:
https://stackoverflow.com/questions/8853396/logical-operator-in-a-handlebars-js-if-conditional

```
{{#ifCond expression}}
Value to show if true case
{{else}}
Value to show if false case
{{/ifCond}}
```

I have checked the examples, but I always go the same lifetime problem, there is my code:
```
fn ifcond(
    h: &Helper,
    handle: &Handlebars,
    ctx: &Context,
    render_ctx: &mut RenderContext,
    out: &mut dyn Output,
) -> Result<(), RenderError> {
    let cond = h
        .param(0)
        .and_then(|ref v| v.value().as_bool())
        .ok_or(RenderError::new(
            "Ifcond takes a boolean !",
        ))? as bool;
    let temp = if cond
    {
        h.template()
    }
    else
    {
        h.inverse()
    };
    match temp
    {
        Some(t) =>
        {
            t.render(handle, ctx, render_ctx, out)
        },
        None => Ok(()),
    }

}
```

I always got this error:
```
error[E0623]: lifetime mismatch
   --> src/helpers.rs:444:35
    |
413 |     h: &Helper,
    |         ------ these two types are declared with different lifetimes...
...
416 |     render_ctx: &mut RenderContext,
    |                      -------------
...
444 |             t.render(handle, ctx, render_ctx, out)
    |                                   ^^^^^^^^^^ ...but data from `h` flows into `render_ctx` here

error: aborting due to previous error
```

I don't understand because I don't see any differences between the examples I found and my code.

I use handlebars 2.0.1.

Thank you !

NB: There is a way to do this with the handlebars_helper! macro?

**Comment by sunng87** [comment]
@crone123 In this case you will need to explicit lifetime on function arguments. Check this case for solution:

https://github.com/sunng87/handlebars-rust/blob/master/tests/helper_function_lifetime.rs

**Comment by crone123** [comment]
Ok, it works !!!
Really, a very big thanks to you !!! :D

**Comment by sunng87** [comment]
Thanks! Helper API is at its relatively early stage and may receive aggressive changes in future. Just let me know if you have questions using the API.

**Comment by crone123** [comment]
Ok, I think, a full example like the file you create for me will be welcome on the doc page for helpers, because we found people who make code like this on internet, but it was for an older version of this lib who doesn't work anymore.

But, really, good job, I discovered handlebars with rocket.rs and with your library, and except for one or two things not clear directly, you made it really easy to use, and fast to run.

I'm using Rust for only 6 months, I love this language (it's now my main programming language), but lifetime are sometimes difficult to use. I understood the concept, but for the moment I don't know really where I need to add a lifetime or not.

Thank you again for your precious help ! :)
